h/views/api/flags.py

```diff
@@ -18,19 +19,23 @@
 def create(context, request):
     request.find_service(name="flag").create(request.user, context.annotation)
 
-    _email_group_admin(request, context.annotation)
+    _email_group_moderators(request, context.annotation)
 
     return HTTPNoContent()
 
 
-def _email_group_admin(request, annotation):
+def _email_group_moderators(request, annotation):
     incontext_link = links.incontext_link(request, annotation)
     if incontext_link is None:
         incontext_link = annotation.target_uri
 
-    group = annotation.group
-    if group.creator and group.creator.email:
-        send_params = flag_notification.generate(
-            request, group.creator.email, incontext_link
-        )
-        mailer.send.delay(*send_params)
+    group_members_service = request.find_service(name="group_members")
+
+    memberships = group_members_service.get(
+        annotation.group, roles=list(GROUP_MODERATE_PREDICATES.keys())
+    )
```

Comment:
I originally wanted this to go through the permissions system to get the list of users who have the `Permission.Group.MODERATE` permission in the context of this annotation's group: create a `context` object for the group, then for each member create an `identity` object and call [`identity_permits(identity, context, Permission.Group.MODERATE)`](https://github.com/hypothesis/h/blob/8765297d0f6c1c3df82d8d64087dd2663d77ccb1/h/security/permits.py#L9-L31) to see if that member should get a moderation email. But there are two problems with that:

1. We'd have to fetch all the group's members from the DB and iterate over them, doing a permissions check for each individual member to find out which ones to email
2. Creating an identity for a user with `Identity.from_models(user)` [iterates over all the user's memberships](https://github.com/hypothesis/h/blob/8765297d0f6c1c3df82d8d64087dd2663d77ccb1/h/security/identity.py#L65) (of all groups that the user is a member of). So we don't want to iterate over all a group's memberships calling `Identity.from_models(membership.user)`  for each of them

I think we want to avoid a solution that scales poorly as the number of plain members (not even moderators) of the group increases, so I think the solution in this PR that duplicates some logic is better.

Reply:
The `list()` is actually need to get an `assert_called_once_with()` in the tests to pass.